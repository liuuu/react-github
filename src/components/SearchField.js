import React from "react";
import { Input, Icon } from "antd";

const SearchField = props =>
  <Input
    style={{ display: "inline-block", width: "40%", marginTop: 30 }}
    addonBefore={<Icon type="search" />}
    size="large"
    placeholder="filter the projects"
    value={props.value}
    onChange={props.handleChange}
  />;

export default SearchField;
