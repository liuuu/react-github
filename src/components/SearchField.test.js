import React from "react";
import SearchField from "./SearchField";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import renderer from "react-test-renderer";

it("SearchField snapshot", () => {
  const tree = renderer.create(<SearchField />).toJSON();
  expect(tree).toMatchSnapshot();
});

const handleChange = jest.fn();
it("change should have value as props passed", () => {
  const component = mount(
    <SearchField handleChange={handleChange} value="inputTesting" />
  );
  expect(component.find("input").node.value).toBe("inputTesting");
});

it("change should invoke the handleChange function", () => {
  const component = mount(
    <SearchField handleChange={handleChange} value="inputTesting" />
  );
  component.find("input").simulate("change");
  expect(handleChange).toHaveBeenCalledTimes(1);
  expect(handleChange).toHaveBeenCalled();
});
