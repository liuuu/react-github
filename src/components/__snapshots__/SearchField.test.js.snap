// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField snapshot 1`] = `
<span
  className="ant-input-group-wrapper"
  style={
    Object {
      "display": "inline-block",
      "marginTop": 30,
      "width": "40%",
    }
  }
>
  <span
    className="ant-input-wrapper ant-input-group"
  >
    <span
      className="ant-input-group-addon"
    >
      <i
        className="anticon anticon-search"
      />
    </span>
    <input
      className="ant-input ant-input-lg"
      disabled={false}
      onChange={undefined}
      onKeyDown={[Function]}
      placeholder="filter the projects"
      size="large"
      style={null}
      type="text"
      value=""
    />
  </span>
</span>
`;
